### PR TITLE
Java/C#: Generalize script for generating flow models.

### PR DESCRIPTION
**Pull request by michaelnebel**
In this PR, we move the *generic* parts of GenerateFlowModel into a separate class, which can then be used for multiple languages.
At the time of writing the *last* commit of the PR, then imports the new *generic* module and instantiates a new object, which is then used for model generation.
My experience with python is limited, but it seems like a bit of hack to modify *sys.path* before importing the other python module.
@tamasvajk : Do you know a better way of achieving the same?

I suppose one option is to move actual generation into the generate-flow-models.py file and just parameterise the script with "language". If we want a script in the local folder, we can just make a symlink to the shared generator script.

### CODE REVIEW

**Comment by michaelnebel** [comment]
I suppose one option is to move *actual* generation into the generate-flow-models.py file and just parameterise the script with "language". If we want a script in the local folder, we can just make a symlink to the shared generator script.

**Review by tamasvajk — COMMENTED** [review]
Added some minor comments. It looks plausible, but I haven't yet checked the model generation, so I have no clear view of the process.

I don't know if you know about this, but if you want to test this in an action, you can push it to a repo in the https://github.com/dsp-testing org.

**Comment by michaelnebel** [comment]
> Added some minor comments. It looks plausible, but I haven't yet checked the model generation, so I have no clear view of the process.
> 
> I don't know if you know about this, but if you want to test this in an action, you can push it to a repo in the https://github.com/dsp-testing org.

It is probably easiest, if you get an introduction during this week, if we should talk about some of the other implementation details as well.
For now, the only *real* framework, where the models has been successfully created for C# is efcore (this is something I am doing manually as it requires parts of several different unmerged PRs). In any case, for the changes on this PR, the Models as Data actions for java appears to still be working and the *artifacts* are still there as well (ie. files containing the generated code). We can also touch upon that when walking through the code.

File: java/ql/src/utils/model-generator/GenerateFlowModel.py [hunk 1]
```diff
@@ -1,178 +1,15 @@
...
-
-qllContents = qllTemplate.format(shortname, sinkCsv, sourceCsv, summaryCsv)
-
 
-with open(frameworkTarget, "w") as frameworkQll:
-    frameworkQll.write(qllContents)
+# Add Model as Data script directory to sys.path.
+gitroot = subprocess.check_output(["git", "rev-parse", "--show-toplevel"]).decode("utf-8").strip()
+madpath = os.path.join(gitroot, "misc/scripts/models-as-data/")
+sys.path.append(madpath)
```
[the remark this answers is not in the document]
Comment (michaelnebel) [reply java/ql/src/utils/model-generator/GenerateFlowModel.py:10]: We might have to live with the uglyness in the local script. IMO it is nice to have the actual script we execute next to the other stuff, such that it is easier to find.

**Comment by michaelnebel** [comment]
@tamasvajk : You are probably the best candidate for review'ing.
In any case the CI doesn't fail which at least indicates that nothing is broken with this change.

Let me know, if you have any more comments that needs to be incorporated before merging.

**Pull request merged**